examples/siro_sandbox/sandbox_app.py

```diff
@@ -334,11 +361,12 @@ def sim_update(self, dt):
         )
         post_sim_update_dict["cam_transform"] = cam_transform
 
-        post_sim_update_dict[
-            "keyframes"
-        ] = (
+        keyframes = (
             self.get_sim().gfx_replay_manager.write_incremental_saved_keyframes_to_string_array()
         )
+        post_sim_update_dict["keyframes"] = keyframes
+        if self._enable_gfx_replay_save:
+            self._recording_keyframes.append(keyframes[0])
```

Comment:
How about assert that `keyframes` length is 1 (or better yet, just append all the frames in `keyframes`)?